sklearn/utils/validation.py

```diff
@@ -1946,7 +1946,7 @@ def _check_response_method(estimator, response_method):
     return prediction_method
 
 
-def _check_fit_params(X, fit_params, indices=None):
+def _check_method_params(X, fit_params, indices=None):
```

Comment:
If I understand the change here, we want to use this method to validate parameters but not only in `fit`.
It is a bit weird to not rename `fit_params` if this is the case.

Reply:
This was just a renaming of a method and all the other occurrences in this PR are just its usages.

Reply:
@adrinjalali I think you have also made this change in the pipeline PR?

Reply:
Do you think this renaming and all the corresponding changes would be worth doing in a follow up PR to avoid any confusions in this one?

Reply:
I think we can change the param name here. The pipeline PR isn't touching this file.